### PR TITLE
Mailchimp incremental fixes

### PR DESCRIPTION
**Pull request by sherifnada**
## What
* Fixes incorrect passing of kwargs in python
* Correctly formats iso strings when being stored back in the state object

### CODE REVIEW

File: airbyte-integrations/connectors/source-mailchimp/sample_files/state.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "Campaigns": {"create_time": "2020-11-23T05:42:11+00:00"},
```
Comment (cgardens) [inline airbyte-integrations/connectors/source-mailchimp/sample_files/state.json:2]: is it really `create_time` and not `created_time`?

File: airbyte-integrations/connectors/source-mailchimp/source_mailchimp/client.py [hunk 2]
```diff
@@ -63,19 +63,25 @@ def lists(self, state: DefaultDict[str, any] = None) -> Generator[AirbyteMessage
...
+        else:
+            max_date_created = None
+
         offset = 0
-        max_date_created = date_created
         done = False
         while not done:
-            lists_response = self._client.lists.all(dict(default_params, count=self.PAGINATION, offset=offset))["lists"]
+            params = dict(default_params, count=self.PAGINATION, offset=offset)
+            lists_response = self._client.lists.all(**params)["lists"]
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-mailchimp/source_mailchimp/client.py:77]: when passing a dictionary and you want it to be parsed as keyword args, it must be preceded by `**`

File: airbyte-integrations/connectors/source-mailchimp/source_mailchimp/client.py [hunk 3]
```diff
@@ -63,19 +63,25 @@ def lists(self, state: DefaultDict[str, any] = None) -> Generator[AirbyteMessage
         stream_name = self._LISTS
         date_created = self._get_cursor_or_none(state, stream_name, cursor_field)
 
-        default_params = {"since_date_created": date_created, "sort_field": cursor_field, "sort_dir": "ASC"}
+        default_params = {"sort_field": cursor_field, "sort_dir": "ASC"}
```
Comment (cgardens) [inline airbyte-integrations/connectors/source-mailchimp/source_mailchimp/client.py:66]: can you put lines 66 through 71 in a function and reuse it for campaigns?

**Review by cgardens — APPROVED** [review]

File: airbyte-integrations/connectors/source-mailchimp/source_mailchimp/client.py (hunk 3 above)
Comment (cgardens) [reply airbyte-integrations/connectors/source-mailchimp/source_mailchimp/client.py:66]: don't need to do this now if you're still iterating and figuring stuff out.

File: airbyte-integrations/connectors/source-mailchimp/sample_files/state.json (hunk 1 above)
Comment (sherifnada) [reply airbyte-integrations/connectors/source-mailchimp/sample_files/state.json:2]: yup!

File: airbyte-integrations/connectors/source-mailchimp/source_mailchimp/client.py (hunk 3 above)
Comment (sherifnada) [reply airbyte-integrations/connectors/source-mailchimp/source_mailchimp/client.py]: DRY'd it into a helper...unclear if it's more readable? it's certainly less repeated code.

**Pull request merged**